api/rbdigital.py

```diff
@@ -742,10 +751,23 @@ def _find_or_create_remote_account(self, patron):
         # authorization identifier. And there is no preexisting
         # Credential representing a dummy account, or this method
         # wouldn't have been called. We must create a new account.
-        return self.create_patron(
-            patron.library, patron.authorization_identifier,
-            self.patron_email_address(patron)
-        )
+        try:
+            return self.create_patron(
+                patron.library, patron.authorization_identifier,
+                self.patron_email_address(patron)
+            )
+        except RemotePatronCreationFailedException:
+            # Its possible to get a RemotePatronCreationFailedException if an account
+            # was already created for this patron, but never put in the DB due to an
+            # error. Here we try to recover that account using its email address.
+            remote_identifier = self.patron_remote_identifier_lookup(
```

Comment:
This isn't working right now. I think it's a RBDigital API issue where they can't lookup a patron with a `+` in their email address. Waiting to hear back from RBDigital about it.